youtube_dl/extractor/fivemin.py

```diff
@@ -59,26 +106,39 @@ def _real_extract(self, url):
             'https://syn.5min.com/handlers/SenseHandler.ashx?' + query,
             video_id)
         if not response['success']:
-            err_msg = response['errorMessage']
-            if err_msg == 'ErrorVideoUserNotGeo':
-                msg = 'Video not available from your location'
-            else:
-                msg = 'Aol said: %s' % err_msg
-            raise ExtractorError(msg, expected=True, video_id=video_id)
+            raise ExtractorError(
+                    '%s said: %s' % (
+                        self.IE_NAME,
+                        self._ERRORS.get(response['errorMessage'], response['errorMessage'])),
+                    expected=True)
         info = response['binding'][0]
 
-        second_id = compat_str(int(video_id[:-2]) + 1)
         formats = []
-        for quality, height in [(1, 320), (2, 480), (4, 720), (8, 1080)]:
-            if any(r['ID'] == quality for r in info['Renditions']):
+        video_url = compat_parse_qs(compat_urllib_parse_urlparse(info['EmbededURL']).query)['videoUrl'][0]
+        qs = compat_urllib_parse_urlparse(video_url).query
+        if qs:
+            qs = '?' + qs
+        for rendition in info['Renditions']:
+            if rendition['RenditionType'] == 'm3u8':
+                formats.extend(self._extract_m3u8_formats(rendition['Url'], video_id, m3u8_id='hls'))
+            elif rendition['RenditionType'] == 'aac':
+                continue
+            else:
+                rendition_url = video_url[:10] + video_url[10:].replace('//', '/%s/' % rendition['ID'])
+                rendition_url = rendition_url[:rendition_url.rindex('.')] + '.' + rendition['RenditionType'] + qs
```

Comment:
I think this two lines are better using  compat_urllib_parse_urlparse and the `_replace` method:
* It's not immediately clear clear why you use `video_url[:10]`, it's better if you explicitly change the path
* `rendition_url.rindex('.')` won't works as expected if the query contains a dot, and normal `index` will match the dot from the domain.